### ISSUE TITLE
Exception message with unicode characters in nested steps 

### ISSUE THREAD

**Issue by yucer** [behave/behave#424]
Take the nested step sample:

https://jenisys.github.io/behave.example/tutorials/tutorial08.html

With this nested step implementation:

```python
@when('I press the big red button')
def step_impl(context):
    assert False, 'Ungültiger Wert'
    context.red_button_pressed += 1

``` 
And the unicode exception is raised when the step is used as nested step:

```cucumber
Feature: Step executes other Steps (tutorial08) # tutorial05_step_data.feature:2

  Scenario: Step by Step            # tutorial05_step_data.feature:4
    Given I start a new game        # steps/step_tutorial05.py:7 0.000s
    When I press the big red button # steps/step_tutorial05.py:12 0.000s
      Assertion Failed: UngÃ¼ltiger Wert

    And I duck                      # None
    Then I reach the next level     # None

  Scenario: Execute multiple Steps in middle Step  # tutorial05_step_data.feature:10
    Given I start a new game                       # steps/step_tutorial05.py:7 0.000s
    When I do the same thing as before             # steps/step_tutorial05.py:21 0.001s
      Assertion Failed: 'ascii' codec can't encode characters in position 84-85: ordinal not in range(128)

    Then I reach the next level                    # None

```
It's like it can not redirect properly the exception output from the nested step.

**Comment by jenisys** [comment]
I am already looking into this problem. Python3 does not have this problem. Therefore, if you have the possibility to switch to Python 3.5.1, do it (until the problem is solved). Otherwise, try to avoid causing Unicode problems.

**Comment by yucer** [comment]
Actually not. I'm using https://github.com/camptocamp/oerpscenario.

It uses an xml-rpc client to connect to an odoo server and test it. The python modules of the server are loaded in memory alongside with the client. When an error ocurrs, you can trace with pdb all the stack trace from the client to the server.

As long as I know odoo is not compatible with python3. So I suppose that it is to risky.

Anyway, I will try to search for the problem and If I found it I'll propose a solution.

**Comment by yucer** [comment]
Sorry I could not find the exact reason of the failing sample.

I found this useful function to Unicode conversion stuff: [ustr](http://bit.ly/1VR0iIs)

Maybe it could be useful to you in some place.

**Comment by jenisys** [comment]
@yucer Thanks, I will look into that (but note that is similar with the intention I had w/ ``behave.textutil.text()``).

**Comment by jenisys** [comment]
FIXED IN: behave-1.2.6.dev0
